polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -1217,9 +1308,16 @@ async fn handle_our_view_change<Context>(
 	let added: Vec<_> = view.iter().filter(|h| !implicit_view.contains_leaf(h)).collect();
 
 	for leaf in added {
-		let claim_queue = fetch_claim_queue(ctx.sender(), *leaf).await?;
+		let claim_queue: ClaimQueueSnapshot = fetch_claim_queue(ctx.sender(), *leaf).await?;
 		state.per_relay_parent.insert(*leaf, PerRelayParent::new(para_id, claim_queue));
 
+		if let Err(err) =
+			process_block_events(ctx, &mut state.collation_tracker, *leaf, para_id, &state.metrics)
+				.await
+		{
+			gum::debug!(target: LOG_TARGET_STATS, ?err,  "Failed to process block events");
```

Comment:
This will never happen, or I am missing something? (see previous comment)

Reply:
Yeah, you are right!